### ISSUE TITLE
Bug: Markdown renderer removes extra line breaks for Card Description

### ISSUE THREAD

**Issue by asaadmahmood** [mattermost/focalboard#878]
**Summary:**
Bug: Markdown renderer removes extra line breaks for Card Description

**Steps to reproduce the behavior:**
1. Create a card
2. Add two lines of text in the description
3. Give 10 empty lines breaks in between them
4. Click outside
5. The spacing is removed

**Expected behavior:**
The spacing should not be removed.

**Screenshots:**
Attached a video.
https://user-images.githubusercontent.com/11034289/128041967-8a75785a-381b-4970-9dd2-c51d24b16b22.mov

**Platform:**
 - Chrome Version 92.0.4515.107 (Official Build) (x86_64)
 - Mac Os 11.4 (20F71)

**Comment by asaadmahmood** [comment]
@chenilim Not a css issue, checked that.

**Comment by kamre** [comment]
The reason is how `marked.js` library interprets line breaks. It tries to follow github markdown, and on github in comments it looks like this:
![GH-878-Markdown-Line-Breaks](https://user-images.githubusercontent.com/5587620/128124450-7b5073c9-bc8f-4f79-af91-fbe89251e73e.gif)
May be some preprocessing of the markdown text passed into `marked.js` can be done to preserve like breaks.

**Comment by asaadmahmood** [comment]
@kamre Yup, I can see that, our case is probably a bit different, so we may need to preserve line breaks as we would follow a document type example, especially when we introduce the full page layout.

**Comment by kamre** [comment]
There is https://github.com/mattermost/marked
May be something was already done to preserve line breaks in the fork? @hmhealey do have any idea about this?

**Comment by hmhealey** [comment]
I think the `breaks` option for marked might be what you're looking for. I thought we used it in the web app, but the fork is pretty out of date and we have a lot of CSS messing with Markdown there as well, so it might not have needed it.

**Comment by kamre** [comment]
@hmhealey `breaks` [is already set](https://github.com/mattermost/focalboard/blob/main/webapp/src/utils.ts#L163) to `true` and this doesn't produce the desired behaviour because the sequence of line ends is collapsed to single one.

And `focalboard` [is not using forked version](https://github.com/mattermost/focalboard/blob/main/webapp/package.json#L39). That is why I have asked if using forked version can help for this issue.

**Comment by hmhealey** [comment]
Oh, I see. I misread the ticket description.

In that case, it's just part of Markdown where it doesn't follow the user's inputted line breaks unless they're in code blocks. Even the web app's version of marked doesn't preserve them despite it predating the CommonMark spec.
